Armstrong_number.py

```diff
@@ -1,21 +1,24 @@
-def is_armstrong_number(number):
-    total = 0
+"""
+In number theory, a narcissistic number (also known as a pluperfect digital invariant (PPDI), an Armstrong number (after Michael F. Armstrong) or a plus perfect number), 
+in a given number base b, is a number that is the total of its own digits each raised to the power of the number of digits.
+Source: https://en.wikipedia.org/wiki/Narcissistic_number
+NOTE:
+this scripts only works for number in base 10
+"""
 
-    # find the sum of the cube of each digit
-    temp = number
-    while temp > 0:
-        digit = temp % 10
-        total += digit ** 3
-        temp //= 10
+def is_armstrong_number(number:str):

```

Comment:
The function could be slower than the older implementation. Due to the `for-loops`. I rather go with `1 while-loop` than `2 for-loops`. It would be much faster if you approach the problem with list comprehensions and the `sum()`

```python

total:int = sum( [int(digit) ** exp for digit in number] )

# display the result
...
...
...
```

Reply:
So while loop is faster then for loop?